system/Test/FilterTestTrait.php

```diff
@@ -0,0 +1,289 @@
+
+				$routes->getRoutes('*'); // Triggers discovery
+				$this->collection = $routes;
+			}
+
+			$this->doneFilterSetUp = true;
+		}
+	}
+
+	//--------------------------------------------------------------------
+	// Utility
+	//--------------------------------------------------------------------
+
+	/**
+	 * Returns a callable method for a filter position
+	 * using the local HTTP instances.
+	 *
+	 * @param FilterInterface|string $filter   The filter instance, class, or alias
+	 * @param string                 $position "before" or "after"
+	 *
+	 * @return callable
+	 */
+	protected function getFilterCaller($filter, string $position): callable
+	{
+		if (! in_array($position, ['before', 'after'], true))
+		{
+			throw new InvalidArgumentException('Invalid filter position passed: ' . $position);
+		}
+
+		if (is_string($filter))
+		{
+			// Check for an alias (no namespace)
+			if (strpos($filter, '\\') === false)
+			{
+				if (! isset($this->filtersConfig->aliases[$filter]))
+				{
+					throw new RuntimeException("No filter found with alias '{$filter}'");
+				}
+
+				$filter = $this->filtersConfig->aliases[$filter];
+			}
+
+			// Get an instance
+			$filter = new $filter();
+		}
+
+		if (! $filter instanceof FilterInterface)
+		{
+			throw FilterException::forIncorrectInterface(get_class($filter));
+		}
+
+		if ($position === 'before')
+		{
+			$request = clone $this->request;
+
+			return function (array $params = null) use ($filter, $request) {
+				return $filter->before($request, $params);
+			};
+		}
+
+		$request  = clone $this->request;
+		$response = clone $this->response;
+
+		return function (array $params = null) use ($filter, $request, $response) {
+			return $filter->after($request, $response, $params);
+		};
+	}
+
+	/**
+	 * Gets an array of filter aliases enabled
+	 * for the given route at position.
+	 *
+	 * @param string $route    The route to test
+	 * @param string $position "before" or "after"
+	 *
+	 * @return string[] The filter aliases
+	 */
+	protected function getFiltersForRoute(string $route, string $position): array
+	{
+		$this->filters->reset();
+
+		if ($routeFilter = $this->collection->getFilterForRoute($route))
+		{
+			$this->filters->enableFilter($routeFilter, $position);
+		}
+
+		$aliases = $this->filters->initialize($route)->getFilters();
+
+		$this->filters->reset();
+		return $aliases[$position];
+	}
+
+	//--------------------------------------------------------------------
+	// Assertions
+	//--------------------------------------------------------------------
+
+	/**
+	 * Asserts that the given route at position uses
+	 * the filter (by its alias).
+	 *
+	 * @param string $route    The route to test
+	 * @param string $position "before" or "after"
+	 * @param string $alias    Alias for the anticipated filter
+	 *
+	 * @return void
+	 */
+	protected function assertFilter(string $route, string $position, string $alias): void
+	{
+		if (! in_array($position, ['before', 'after'], true))
```

Comment:
move this check to getFiltersForRoute so that other assertions can check the value of position

Reply:
Ah thanks I thought I had done this. I seem to have crossed commits at one point because my Rector changes reverted so maybe I reverted that too? Fixing.